examples/cpp/hello_tf.cpp

```diff
@@ -0,0 +1,130 @@
+    cout << "Backend: " << backend_name << std::endl;
+  }
+}
+
+// Create a simple computation graph and run
+void RunSimpleNetworkExample() {
+  // Create the graph
+  tensorflow::Scope root = tensorflow::Scope::NewRootScope();
+  // Matrix A = [3 2; -1 0]
+  auto A = tensorflow::ops::Const(root, {{0.03f, 0.022f}, {-0.001f, 0.025f}});
+  // Vector b = [3 5]
+  auto b = tensorflow::ops::Const(root, {{0.345f, 0.35f}});
+  // v = Ab^T
+  auto v = tensorflow::ops::MatMul(root.WithOpName("v"), A, b,
+                                   tensorflow::ops::MatMul::TransposeB(true));
+  // R = softmax(v)
+  auto R = tensorflow::ops::Softmax(root, v);
+
+  // Turn off optimizations so that all the nodes are processed
+  tensorflow::SessionOptions options;
+  options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(tensorflow::OptimizerOptions_Level_L0);
+  options.config.mutable_graph_options()
+      ->mutable_rewrite_options()
+      ->set_constant_folding(tensorflow::RewriterConfig::OFF);
+
+  if (tensorflow::ngraph_bridge::ngraph_tf_is_grappler_enabled()) {
+    auto* custom_config = options.config.mutable_graph_options()
+                              ->mutable_rewrite_options()
+                              ->add_custom_optimizers();
+
+    custom_config->set_name("ngraph-optimizer");
+    (*custom_config->mutable_parameter_map())["ngraph_backend"].set_s("CPU");
+    (*custom_config->mutable_parameter_map())["device_id"].set_s("1");
+
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_min_graph_nodes(-1);
+
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_meta_optimizer_iterations(tensorflow::RewriterConfig::ONE);
+  }
+
+  tensorflow::ClientSession session(root, options);
+
+  std::vector<tensorflow::Tensor> outputs;
+  session.Run({R}, &outputs);
+
+  // Print the output
+  std::cout << "Result: " << outputs[0].matrix<float>() << std::endl;
+}
+
+void PrintVersion() {
+  // nGraph Bridge version info
+  std::cout << "Bridge version: "
+            << tensorflow::ngraph_bridge::ngraph_tf_version() << std::endl;
+  std::cout << "nGraph version: "
+            << tensorflow::ngraph_bridge::ngraph_lib_version() << std::endl;
+  std::cout << "CXX11_ABI Used: "
+            << tensorflow::ngraph_bridge::ngraph_tf_cxx11_abi_flag()
+            << std::endl;
+  std::cout << "Grappler Enabled? "
+            << (tensorflow::ngraph_bridge::ngraph_tf_is_grappler_enabled()
+                    ? std::string("Yes")
+                    : std::string("No"))
+            << std::endl;
+  std::cout << "Variables Enabled? "
+            << (tensorflow::ngraph_bridge::ngraph_tf_are_variables_enabled()
+                    ? std::string("Yes")
+                    : std::string("No"))
+            << std::endl;
+
+  PrintAvailableBackends();
+}
+
+int main(int argc, char** argv) {
+  PrintVersion();
```

Comment:
If we mean this app to run only on grapple build, we should assert tensorflow::ngraph_bridge::ngraph_tf_is_grappler_enabled()

Reply:
Actually, this is made sure while building the app. The app's build rule has `ngraph_bridge_lib` as a dependency which has `NGRAPH_TF_USE_GRAPPLER_OPTIMIZER` specified as the build flag. So, we don't need the assert.